kong/tools/wrpc.lua

```diff
@@ -255,23 +253,28 @@ local wrpc_peer = {
 }
 wrpc_peer.__index = wrpc_peer
 
-local function is_wsclient(conn)
-  return conn and not conn.close or nil
-end
+local channels = {}
 
 --- a `peer` object holds a (websocket) connection and a service.
 function wrpc.new_peer(conn, service, opts)
-  opts = opts or {}
-  return setmetatable(merge({
+  local ret = setmetatable({
     conn = conn,
     service = service,
     seq = 1,
-    request_queue = is_wsclient(conn) and Queue.new(),
+    request_queue = Queue.new(),
```

Comment:
the websocket library is different for "client" and "server" connections.  only clients require a transmit thread, and therefore a queue.  in fact,  the presence of the queue is checked later to spawn the transmit thread.  this change means "server" connections would now pass through this queue and thread.

or is it now required because of the "shallow remotes"?

Reply:
It's a mistake. I was trying to let `remote_peer` own its queue, and that solution is abandoned.